metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -568,6 +574,57 @@ final class FileDecoderProvider(
     }
   }
 
+  private def decodeNIR(
+      path: AbsolutePath
+  ): Future[DecoderResponse] = {
+    val scalaNativeCLIDependency =
+      Dependency.of("org.scala-native", "scala-native-cli_2.13", "0.4.10")
```

Comment:
Latest is 0.4.14. This will have to stay pretty up-to-date b/c NIR is continually evolving, I wonder if there is a better way we can manage this version.

Reply:
I guess my concern was having metals depend on the scala-native-cli (or nir) libraries directly. If that's not a concern then this dynamic download is probably not required.

Reply:
Ah, actually I think a dynamic download is _better_ because it would allow the scala-native-cli version to be bumped to latest without requiring a new release of metals.

Reply:
I'm curious if I can:

1. add this as an optional dep so steward can do the update PRs. 
2. Pass the version in with build info
3. Default a config to that version
4. Use that condid here

A few steps but would hopefully balance devex of this code with usability

Reply:
Could we not check the classpath which version of native is being used?

Reply:
I would think so? If there is a way to access the native build target then I presume that can be inspected to determine the native version.

Reply:
We don't have a separate build target data for native (could be done), but we do have a list of jars that is being used, so potentially we could scan it for a particular native jar and check the pom.xml or try to figure out native version from the name (not the greatest, but we can fallback to the newest known version if the approach fails)